### ISSUE TITLE
🐛 noConsole not reporting when using console in arrow function

### ISSUE THREAD

**Issue by AdriVanHoudt** [biomejs/biome#5053]
### Environment information

```block
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v22.13.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.9.2"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

Something as simple as 
```ts
[].forEach(() => console.log("test"))
```
doesn't trigger the noConsole rule.

https://biomejs.dev/playground/?lintRules=all&code=WwBdAC4AZgBvAHIARQBhAGMAaAAoACgAKQAgAD0APgAgAGMAbwBuAHMAbwBsAGUALgBsAG8AZwAoACIAdABlAHMAdAAiACkAKQAKAAoAYwBvAG4AcwB0ACAAeAAgAD0AIAAoACkAIAA9AD4AIABjAG8AbgBzAG8AbABlAC4AbABvAGcAKAAiAHQAZQBzAHQAIgApAA%3D%3D

### Expected result

It creates a diagnostic for the noConsole rule

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
